src/twisted/defer/newsfragments/10222.bugfix

```diff
@@ -0,0 +1,2 @@
+avoid AssertionError in deferred awaited twice when debugging is disabled
+always clear the result of awaited Deferreds
```

Comment:
If there are 2 bugs being fixed here, there should be 2 tickets and 2 news files, the format here is not multi-line.

Reply:
the assertion error in debug mode was actually just fallout from the result not being consumed correctly

Reply:
In that case the feedback is "Make it a full, grammatically correct sentence" :)